net_util/src/mac.rs

```diff
@@ -26,26 +26,28 @@ impl MacAddr {
     {
         let v: Vec<&str> = s.as_ref().split(':').collect();
         let mut bytes = [0u8; MAC_ADDR_LEN];
-        let common_err = Err(io::Error::other(format!(
-            "parsing of {} into a MAC address failed",
-            s.as_ref()
-        )));
+        let common_err = || {
+            Err(io::Error::other(format!(
+                "parsing of {} into a MAC address failed",
+                s.as_ref()
+            )))
+        };
```

Comment:
This change is unrelated to fixing the parsing, and I imagine the compiler is probably smart enough not to construct this until used anyway?